domain-model-assistant/Assets/Components/Scripts/Diagram.cs

```diff
@@ -203,26 +167,39 @@ public void LoadJson(string cdmJson)
     {
         ResetDiagram();
         //Debug.Log(cdmJson);
-        var classDiagram = JsonUtility.FromJson<ClassDiagramDTO>(cdmJson);
+        var cdmDto = JsonUtility.FromJson<ClassDiagramDTO>(cdmJson);
 
         //store attributes of class in a dictionary
-        classDiagram.classes.ForEach(cls => this.classIdToAttributes[cls._id] = cls.attributes);
+        cdmDto.classDiagram.classes.ForEach(cls => classIdToAttributes[cls._id] = cls.attributes);
         //store attribute types. Map type id to eclass tye
-        classDiagram.types.ForEach(type =>
+        cdmDto.classDiagram.types.ForEach(type =>
         {
             //cache eClass attr with shortened substring
             //Eg. http://cs.mcgill.ca/sel/cdm/1.0#//CDString -> string
-            string res = type.eClass.Substring(type.eClass.LastIndexOf('/') + 1).Replace("CD", "").ToLower();
+            string res = type.eClass[(type.eClass.LastIndexOf('/') + 1)..].Replace("CD", "").ToLower();
             if (!attrTypeIdsToTypes.ContainsKey(type._id))
             {
-                this.attrTypeIdsToTypes[type._id] = res;
+                attrTypeIdsToTypes[type._id] = res;
             }
         });
         // maps each _id to its (class object, position) pair 
         var idsToClassesAndLayouts = new Dictionary<string, List<object>>();
 
-        classDiagram.classes.ForEach(cls => idsToClassesAndLayouts[cls._id] = new List<object> { cls, null });
-        classDiagram.layout.containers[0].value/*s*/.ForEach(contVal =>
+        cdmDto.classDiagram.classes.ForEach(cls => idsToClassesAndLayouts[cls._id] = new List<object> { cls, null });
+        Debug.Log("cdmJson: " + cdmJson);
+        // Debug.Log("classDiagram: " + classDiagram);
+        // Debug.Log("classDiagram.layout: " + classDiagram.layout);
+        // Debug.Log("JsonUtility.ToJson(classDiagram.layout): " + JsonUtility.ToJson(classDiagram.layout));
+        // Debug.Log("classDiagram.layout.containers: " + classDiagram.layout.containers);
+        if (cdmDto.classDiagram.layout.containers == null)
+        {
+            Debug.Log("classDiagram.layout.containers is null, so early return");
+            return;
+        }
+        Debug.Log("classDiagram.layout.containers.Count: " + cdmDto.classDiagram.layout.containers.Count); // NullReferenceException
+        Debug.Log("classDiagram.layout.containers[0]: " + cdmDto.classDiagram.layout.containers[0]);
+        Debug.Log("classDiagram.layout.containers[0].value: " + cdmDto.classDiagram.layout.containers[0].value);
```

Comment:
All these `Debug.Log`s, including the commented out ones, should be removed once we have more confidence in this logic